pallets/runtime/tests/src/confidential_test.rs

```diff
@@ -73,3 +84,59 @@ fn range_proof_we() {
         true
     );
 }
+
+#[test]
+fn account_create_tx() {
+    ExtBuilder::default()
+        .build()
+        .execute_with(account_create_tx_we);
+}
+
+fn account_create_tx_we() {
+    let alice = AccountKeyring::Alice.public();
+    let alice_id = register_keyring_account(AccountKeyring::Alice).unwrap();
+
+    let valid_asset_ids: Vec<AssetId> = vec![1, 2, 3]
+        .iter()
+        .map(|id| AssetId::from(id.clone()))
+        .collect();
+    Confidential::store_valid_asset_ids(Origin::signed(alice), valid_asset_ids.clone());
+
+    let mut rng = StdRng::from_seed([10u8; 32]);
+    let elg_secret = ElgamalSecretKey::new(Scalar::random(&mut rng));
+    let elg_pub = elg_secret.get_public_key();
+    let enc_keys = EncryptionKeys {
+        pblc: elg_pub.into(),
+        scrt: elg_secret.into(),
+    };
+
+    let sign_keys = MiniSecretKey::from_bytes(&[11u8; 32])
+        .expect("Invalid seed")
+        .expand_to_keypair(ExpansionMode::Ed25519);
```

Comment:
Is it ok for the tests to use rand and generate secrets? Or do you have a different way of mocking the Wallet's behaviour?

Reply:
In general, we are using `substrate-test-utils` pallet which will allow us to use Alice , Bob, Charlie, or different test accounts which allow you to sign the transaction. But you have to check whether it gives you all the utility or not. 

I am not sure what you are referring to generate secrets, If it is a key pair then substrate-test-utils is the best fit.
using rand is fine in test